src/libs/PersonalDetailsUtils.js

```diff
@@ -63,10 +63,22 @@ function getAccountIDsByLogins(logins) {
     return _.reduce(
         logins,
         (foundAccountIDs, login) => {
-            const currentDetail = _.find(personalDetails, (detail) => detail.login === login);
+            let isUserWithNoLogin = false;
+            const currentDetail = _.find(personalDetails, (detail) => {
+                if (!_.isUndefined(detail.login)) {
+                    isUserWithNoLogin = true;
+                }
+                return detail.login === login;
+            });
             if (!currentDetail) {
-                // generate an account ID because in this case the detail is probably new, so we don't have a real accountID yet
-                foundAccountIDs.push(UserUtils.generateAccountID(login));
+                // If the account login does not match login and the account login is undefined, we check if account display name matches login
```

Comment:
I'm a bit confused about this comment too, cause I think we're talking about logins of multiple accounts as if it were the login of a single account. Can you try to make this a bit clearer/more obvious?